### PR TITLE
TYP/TST: tests/mypy complaints failing locally

### PR DESCRIPTION
**Pull request by jbrockmendel**
This un-xfails-on-mac a test that is passing locally.  Also suppresses some mypy complaints.  I'm still complaints:

```
pandas/io/formats/printing.py:266: error: Incompatible types in assignment (expression has type "Tuple[Type[Dict[Any, Any]]]", base class "BaseFormatter" defined the type as "Type[str]")  [assignment]
pandas/core/dtypes/dtypes.py:865: error: Incompatible types in assignment (expression has type "Callable[[PeriodDtypeBase], int]", base class "PandasExtensionDtype" defined the type as "Callable[[PandasExtensionDtype], int]")  [assignment]
```

that I haven't found a solution for

### CODE REVIEW

File: pandas/tests/io/xml/test_to_xml.py [hunk 1]
```diff
@@ -991,7 +991,9 @@ def test_stylesheet_file_like(datapath, mode):
 def test_stylesheet_io(datapath, mode):
     xsl_path = datapath("io", "data", "xml", "row_field_output.xsl")
 
-    xsl_obj: BytesIO | StringIO
+    # note: By default the bodies of untyped functions are not checked,
+    # consider using --check-untyped-defs
+    xsl_obj: BytesIO | StringIO  # type: ignore[annotation-unchecked]
```
Comment (twoertwein) [inline pandas/tests/io/xml/test_to_xml.py:996]: `def test_stylesheet_io(datapath, mode) -> None:` for mypy to check this function and not emit this note

Comment (jbrockmendel) [reply pandas/tests/io/xml/test_to_xml.py:996]: That works for test_dirname_mixin, for the others produces `error: Untyped decorator makes function "test_stylesheet_io" untyped  [misc]`

Comment (twoertwein) [reply pandas/tests/io/xml/test_to_xml.py:996]: Not sure whether the following is worth it: could lie about `skip_if_no`

`def skip_if_no(package: str, min_version: str | None = None): -> Callable[[F], F]: ...`

A simpler version could just be to remove those `xsl_obj: BytesIO | StringIO` lines - they are not used by mypy/pyright anyways.

Comment (jbrockmendel) [reply pandas/tests/io/xml/test_to_xml.py:996]: hopefully this can be handled correctly upstream https://github.com/pytest-dev/pytest/issues/7469

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel (can follow up with the other typings maybe if it makes things simpler)

**Pull request merged**